test/python/synthesis/test_mcx_synthesis.py

```diff
@@ -0,0 +1,123 @@
+
+        for gate_circuit in [
+            synth_mcx_1_clean_kg24(num_ctrl_qubits),
+            synth_mcx_1_dirty_kg24(num_ctrl_qubits),
+        ]:
+            with self.subTest(gate_circuit=gate_circuit):
+                qc = QuantumCircuit(num_ctrl_qubits + 2)
+                qc.x(list(range(num_ctrl_qubits)))
+                qc.compose(gate_circuit, inplace=True)
+                statevector = Statevector(qc).data
+
+                corrected = np.zeros(2 ** (num_ctrl_qubits + 1), dtype=complex)
+                for i, statevector_amplitude in enumerate(statevector):
+                    i = int(bin(i)[2:].zfill(qc.num_qubits)[1:], 2)  # 1 ancilla qubit
+                    corrected[i] += statevector_amplitude
+                statevector = corrected
+            np.testing.assert_array_almost_equal(statevector.real, reference)
+
+    @data(3, 5, 8, 10, 15)
+    def test_synth_mcx_2_ancilla(self, num_ctrl_qubits):
+        """Test mcx synthesis with 2 ancilla."""
+        reference = np.zeros(2 ** (num_ctrl_qubits + 1))
+        reference[-1] = 1
+
+        for gate_circuit in [
+            synth_mcx_2_clean_kg24(num_ctrl_qubits),
+            synth_mcx_2_dirty_kg24(num_ctrl_qubits),
+        ]:
+            with self.subTest(gate_circuit=gate_circuit):
+                qc = QuantumCircuit(num_ctrl_qubits + 3)
+                qc.x(list(range(num_ctrl_qubits)))
+                qc.compose(gate_circuit, inplace=True)
+                statevector = Statevector(qc).data
+
+                corrected = np.zeros(2 ** (num_ctrl_qubits + 1), dtype=complex)
+                for i, statevector_amplitude in enumerate(statevector):
+                    i = int(bin(i)[2:].zfill(qc.num_qubits)[2:], 2)  # 2 ancilla qubits
+                    corrected[i] += statevector_amplitude
+                statevector = corrected
+            np.testing.assert_array_almost_equal(statevector.real, reference)
+
+    @data(3, 5, 10, 15)
+    def test_synth_mcx_clean_kg24_count(self, num_ctrl_qubits):
+        """Test if cx count of the synth_mcx_clean_kg24 is correct."""
+        for mcx in [
+            synth_mcx_1_clean_kg24(num_ctrl_qubits),
+            synth_mcx_2_clean_kg24(num_ctrl_qubits),
+        ]:
+            with self.subTest(gate_circuit=mcx):
+                tr_mcx = transpile(mcx, basis_gates=["u", "cx"], optimization_level=3)
+                cx_count = tr_mcx.count_ops()["cx"]
+                self.assertLessEqual(cx_count, 12 * num_ctrl_qubits - 18)
+
+    @data(3, 5, 10, 15)
+    def test_synth_mcx_dirty_kg24_count(self, num_ctrl_qubits):
+        """Test if cx count of synth_mcx_dirty_kg24 is correct."""
+        for mcx in [
+            synth_mcx_1_dirty_kg24(num_ctrl_qubits),
+            synth_mcx_2_dirty_kg24(num_ctrl_qubits),
+        ]:
+            with self.subTest(gate_circuit=mcx):
+                tr_mcx = transpile(mcx, basis_gates=["u", "cx"], optimization_level=3)
+                cx_count = tr_mcx.count_ops()["cx"]
```

Comment:
using `optimization_level=3` seems too extensive in this case (as well as in the other cases below)

it's better to do something like:

```
       synthesized_circuit =synth_mcx_1_dirty_kg2(num_ctrl_qubits)
        pm = generate_preset_pass_manager(
            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
        )
        transpiled_circuit = pm.run(synthesized_circuit)
        cx_count = transpiled_circuit.count_ops()["cx"]
```

see some examples in #13961
(you can rebase your PR after we'll merge it).

Reply:
Yes, I think forgot to merge main back. It should be good now.

Reply:
The final tests are in!